sermant-backend/src/main/java/io/sermant/backend/controller/ConfigController.java

```diff
@@ -0,0 +1,160 @@
+import io.sermant.backend.entity.config.Result;
+import io.sermant.backend.entity.config.ResultCodeType;
+import io.sermant.backend.service.ConfigService;
+
+import org.apache.commons.lang.StringUtils;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+import java.util.Optional;
+
+import javax.annotation.Resource;
+
+/**
+ * Config Controller
+ *
+ * @author zhp
+ * @since 2024-05-16
+ */
+@Component
+@RestController
+@RequestMapping("/sermant")
+public class ConfigController {
+    @Resource
+    private ConfigService configService;
+
+    /**
+     * Query Configuration List
+     *
+     * @param configInfo Query criteria for configuration list
+     * @return Configuration List
+     */
+    @GetMapping("/configs")
+    public Result<List<ConfigInfo>> getConfigList(ConfigInfo configInfo) {
+        if (StringUtils.isEmpty(configInfo.getPluginType())) {
+            return new Result<>(ResultCodeType.MISS_PARAM.getCode(), ResultCodeType.MISS_PARAM.getMessage());
+        }
+        Optional<PluginType> optionalPluginType = PluginType.getPluginType(configInfo.getPluginType());
+        if (!optionalPluginType.isPresent()) {
+            return new Result<>(ResultCodeType.FAIL.getCode(), "Invalid plugin name.");
+        }
+        PluginType pluginType = optionalPluginType.get();
+        if (pluginType == PluginType.OTHER) {
+            if (StringUtils.isEmpty(configInfo.getGroup())) {
+                return new Result<>(ResultCodeType.MISS_PARAM.getCode(), ResultCodeType.MISS_PARAM.getMessage());
+            }
+        }
+        return configService.getConfigList(configInfo, pluginType, false);
+    }
+
+    /**
+     * Query Configuration information
+     *
+     * @param configInfo Query criteria for configuration list
+     * @return Configuration information
+     */
+    @GetMapping("/config")
+    public Result<ConfigInfo> getConfig(ConfigInfo configInfo) {
+        if (StringUtils.isEmpty(configInfo.getGroup()) || StringUtils.isEmpty(configInfo.getKey())) {
+            return new Result<>(ResultCodeType.MISS_PARAM.getCode(), ResultCodeType.MISS_PARAM.getMessage());
+        }
+        return configService.getConfig(configInfo);
+    }
+
+    /**
+     * Add configuration
+     *
+     * @param configInfo Configuration information
+     * @return The result of adding configuration
+     */
+    @PostMapping("/config")
+    public Result<Boolean> addConfig(@RequestBody() ConfigInfo configInfo) {
```

Comment:
`()` after `@RequestBody` can be removed

Reply:
done